src/iht.jl

```diff
@@ -245,6 +247,21 @@ function iht_one_step!(v::IHTVariable{T}, x::SnpArray, xbm::SnpBitMatrix, z::Abs
     # update the linear predictors `xb` with the new proposed b, and use that to compute the mean
     update_xb!(v, x, z)
     update_μ!(v.μ, v.xb + v.zc, l)
+   
+    # if Negative Binomial, save previous r and update
+    if typeof(d) == NegativeBinomial{Float64}
+        old_r = d.r
+
+        # Using MM algorithm
+        if est_r == 1
+            d = update_r!(d, y, v.μ)
+
+        # Using Newton's method
+        elseif est_r == 2   
+            new_r = mle_for_θ(y, v.μ, θ=d.r)
+            d = NegativeBinomial(new_r, 0.5)
+        end
+    end
```

Comment:
This presents extra if statement for every iteration of `L0_reg`, even if users do not run negative binomial regression. The easiest solution is to write a specialized `L0_reg` function specifically for negative binomial regression using Julia's multiple dispatch system.